### ISSUE TITLE
Keyboard style Palette?

### ISSUE THREAD

**Issue by annjawn** [Nicejinux/NXDrawKit#24]
Is there an inbuilt way of creating the Palette as a keyboard style? Like the user would be able to dismiss the Palette after selecting the color and continue to draw on the canvas. Currently the Palette kind of takes about half of the screen, but with a keyboard style we can make the whole screen available to the user for drawing and create a better experience. I will try it in my code and submit a PR if you are ok with having a feature like this.

**Comment by Nicejinux** [comment]
Hi @annjawn 
Sure!! Why not!!!!
send me your PR as soon as possible.
Thank you so much!

**Comment by annjawn** [comment]
Here's what I have so far. Also, fancied up the canvas when an image is selected. The canvas will set itself to the size of the image's aspect ratio dynamically, so that after the draw there are no black blank portions left on the top and bottom. Once the "Clear" button is pressed the Canvas will reset to the default size. 👍

It is also possible to hide the palette automatically as the user starts to draw on the canvas. But I felt that it could get annoying for the user so I added a floaty button instead. However, the hide automatically as soon as user draw feature is very easy to implement and uses the `didUpdateDrawing` delegate and can be kept as an option.

P.S: The button says "Down" , I am working on making it dynamic so that it says "Show"/"Hide".
P.S.: I got rid of the RSKImageCropper thing from The Sample project since the Canvas is now capable of resizing itself based on the content so no cropping required. Next, it would be to add zoom to the canvas once an image is added....work in progress.... very cool library by the way. 👌🏼

![nxdraw](https://user-images.githubusercontent.com/26977294/33306843-0b4b5d52-d3c9-11e7-91e4-47b6d3e7963b.gif)
